uktrade/data-hub-frontend
# Refactor typeahead component to become AA compliant

## Description of change
Our current React Typeahead isn't working well against a screen reader and as a result was flagged up by the DAC (Digital Accessibility Centre) in the recent accessibility audit.

## Solution
To correct the typeahead I looked into some research and development that was done by Gov.uk a while ago. They spent a lot of time testing and developing a typeahead/autocomplete with actual users that rely on assistive technologies to browse the web. As a result they did manage to build a component that worked for the users:

Blog article - https://accessibility.blog.gov.uk/2018/05/15/what-we-learned-from-getting-our-autocomplete-tested-for-accessibility/

Result - https://alphagov.github.io/accessible-autocomplete/examples/

In order to get our typeahead working it was a case of updating mark up and adding the right aria attributes. I believe that the changes made in this PR has really improved how the screen readers works against this component.

Note: I have removed the smallTypeahead as we don't use it, this was something I introduced a while ago which was never consumed into the app.

## How to test

1. Run Storybook and look at the “SINGLE SELECT ONLY” (multi-select will be tackled in a slightly different way)
2. Compare the [GDS version](https://alphagov.github.io/accessible-autocomplete/examples/) to ours, look at the mark up and the aria attributes both static and dynamic. 
3. Most importantly run this with a screenreader, if you have a Mac use VoiceOver


[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
